kf_lib_data_ingest/target_apis/kids_first_dataservice.py

```diff
@@ -0,0 +1,768 @@
+"""
```

Comment:
Just going to put this as a general comment up here. The old code used to check which attributes like foreign keys were required (I think using the swagger spec?). If a fk wasn't required it would grab it if it existed and then move on if not. I just ran this on Chung and noticed that it's erroring on building participants because it expects that every participant will have a family id. 

There are probably other cases too so we probably just need to modify the behavior of getting the target id for those fks (e.g. `if CONCEPT.FAMILY.ID in row then get_target_id`) . I don't think we need to use the swagger spec to do this anymore since we're not trying to do anything automatically.

Reply:
> Just going to put this as a general comment up here. The old code used to check which attributes like foreign keys were required.

This is true. I expect that some small updates probably need to be made to the target API classes for things like this (e.g. [] vs .get handling).

> (I think using the swagger spec?)

I don't see anything in the previous swagger usage that does this. I searched the code for both "x-nullable" and "required" and found nothing handling those swagger fields.

Reply:
I'm going to see if changing it so that build_entity is what asserts existence rather than build_key works out advantageously.

Reply:
Ok, can you try the latest commit again?